Blueprint proposal changes

Auwate opened the pull request:
# Description
- Fixes #10

# Changes
- Adds regex
- Adds unit testing

Auwate committed:
Commit:

# Description:
- Added changes necessary to reflect changes in the blueprint proposal

Auwate committed:
Commit:

# Description:
- Added changes necessary to reflect changes in the blueprint proposal

Auwate committed:
Commit:

# Description
- Merge main2.py into main.py

Auwate committed:
Commit:

# Description
- Merge main2.py into main.py

Auwate committed:
Merge branch 'blueprint_proposal_changes' of https://github.com/amlight/kafka_napp into blueprint_proposal_changes

Auwate committed:
Commit:

# Description
- Changing NApp name to kafka_events

Auwate committed:
Commit:

# Changes
- Updated blueprint

Auwate committed:
Commit:

# Changes
- Updated blueprint to use .rst instead of .md

Auwate committed:
Commit:

# Changes
- Updated table syntax to reflect .rst formatting

Auwate committed:
Commit:

# Changes:
- Added Regex
  - Support for all main NApps found on kytos-end-to-end-tests
- Added testing
  - Added unit testing for functionality and expected behavior

Auwate committed:
Commit:

# Changes
- Adding blueprint to keep up with kytos PR

Auwate committed:
Commit:

# Description:
- Remove kytos/sample_Ui.*

Alopalao commented on managers/regex.py:
```diff
@@ -0,0 +1,59 @@
+
+            if rule["type"] == "wildcard":
+                rules.append(re.compile(self._wildcard_to_regex(rule["pattern"])))
+            elif rule["type"] == "match":
```
This part looks great but there is not mentioning of `"match"` anywhere. A quick mention of the possible options for `"type"` in `settings.py` and an explanation of these possible options in `README` will go a long way for anyone that just want to add rules to the pool.

Auwate replied:
Ah you make a great point, thanks Aldo! Originally I had one in for `kytos/mef_eline.created`, but decided to take it out since Vinicius wants us to have all events from all core NApps included anyway.

Alopalao commented on tests/unit/test_regex.py:
```diff
@@ -0,0 +1,77 @@
+        self.assertTrue(self.regex.is_accepted_event("kytos/flow_manager.messages.out.ofpt_barrier_request"))
+        self.assertTrue(self.regex.is_accepted_event("kytos/topology.topology_loaded"))
+        self.assertTrue(self.regex.is_accepted_event("kytos/of_lldp.messages.out.ofpt_packet_out"))
+    {"pattern": "kytos/maintenance.*", "type": "wildcard"}
```
Leftover line here.

Alopalao commented on tests/unit/test_regex.py:
```diff
@@ -0,0 +1,77 @@
+        """
+        self.assertEqual("kytos/maintenance\\..*",
+                         self.regex._wildcard_to_regex("kytos/maintenance.*"))
+        self.assertEqual("kytos/.*", self.regex._wildcard_to_regex("kytos/*"))
```
Add some assert when the result is `false` too.

Alopalao commented on tests/unit/test_regex.py:
```diff
@@ -0,0 +1,77 @@
+        rules: list[Pattern] = self.regex._setup_rules(custom_rules)
+
+        self.assertEqual("kytos/mef_eline\\..*", rules[0].pattern, rules[0])
+        self.assertEqual("kytos/simple_ui.work", rules[1].pattern, rules[1])
```
Some `asserFalse` here to assert all possible outcomes.

Alopalao reviewed (commented):
Good stuff to have a way to add rule matching. It is pretty much done, I left some comments pointing out minor details.

Auwate committed:
Commit:

# Changes
- Removed leftover lines
- Added false checks

Auwate commented:
# Newest Commit
- Removed leftover lines from `test_regex.py`
- Added a test to check for non-membership

Auwate committed:
Commit:

# Description:
- Updated README to include requirements, subscribed events, filtering, and examples

Auwate committed:
Commit:

# Changes
- Added negative testing to test suite

Alopalao commented:
Looks like this PR does not need approval but everything LGTM.
